File: app/src/main/java/com/dpis/module/templates/GlobalPrefillStore.java
package com.dpis.module.templates;

import android.content.SharedPreferences;

public final class GlobalPrefillStore {
    private static final String PREFIX = "default_config.";

    private final SharedPreferences preferences;

    public GlobalPrefillStore(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    public TemplateConfigValue read() {
        return TemplateCustomSemantics.customValue(
                TemplateConfigPreferences.read(preferences, PREFIX));
    }

    public boolean write(TemplateConfigValue value) {
        SharedPreferences.Editor editor = preferences.edit();
        TemplateConfigPreferences.write(editor, PREFIX,
                TemplateCustomSemantics.customValue(value));
        return editor.commit();
    }

    public boolean clear() {
        SharedPreferences.Editor editor = preferences.edit();
        TemplateConfigPreferences.clear(editor, PREFIX);
        return editor.commit();
    }
}

File: app/src/main/java/com/dpis/module/templates/GlobalPrefillStore.kt
package com.dpis.module.templates

import android.content.SharedPreferences

/** Persists the global editor prefill without ever creating a configured package entry. */
class GlobalPrefillStore(
    private val preferences: SharedPreferences,
) {
    fun read(): TemplateConfigValue = TemplateCustomSemantics.customValue(
        TemplateConfigPreferences.read(preferences, PREFIX),
    )

    fun write(value: TemplateConfigValue?): Boolean = preferences.edit().run {
        TemplateConfigPreferences.write(this, PREFIX, TemplateCustomSemantics.customValue(value))
        commit()
    }

    fun clear(): Boolean = preferences.edit().run {
        TemplateConfigPreferences.clear(this, PREFIX)
        commit()
    }

    private companion object {
        const val PREFIX = "default_config."
    }
}
